lib/active_versioning/model.rb

```diff
@@ -0,0 +1,37 @@
+module ActiveVersioning
+  module Model
+    extend ActiveSupport::Concern
+
+    autoload :Versioned,    'active_versioning/model/versioned'
+    autoload :VersionProxy, 'active_versioning/model/version_proxy'
+
+    included do
+      belongs_to :versionable, polymorphic: true
+
+      serialize :object, Hash
+
+      validates :event, presence: true
+
+      scope :newest_first, -> { order created_at: :desc }
+      scope :draft,        -> { where draft: true }
+      scope :committed,    -> { where draft: false }
+
+      def to_s
+        [
+          versionable.to_s,
+          committed_at.try(:utc).try(:strftime, '#%Y%m%d%H%M%S')
+        ].compact.join(' ')
+      end
+
+      def reify
+        resource = versionable(true)
+
+        # Reload the versionable so we get a new one
+        versionable(true)
```

Comment:
This ensures that the `resource` in this case is a different Ruby object than the version's `versionable`.  The result of this method is the object that's modified when you're working with a record's `current_draft`.  Might encounter bugginess if a record and its `current_draft` (just a delegator) point to the same record in memory?  @yaychris can probably explain/confirm as to why this is necessary better than I.